pkg/operation/operation_test.go

```diff
@@ -140,4 +142,43 @@ var _ = Describe("operation", func() {
 			Expect(*o.ShootState.OwnerReferences[0].Controller).To(BeTrue())
 		})
 	})
+
+	Describe("#SaveGardenerResourcesInShootState", func() {
+		var (
+			o                      *Operation
+			ctrl                   *gomock.Controller
+			gardenClient           *mock.MockInterface
+			k8sGardenRuntimeClient *mockclient.MockClient
+		)
+
+		BeforeEach(func() {
+			ctrl = gomock.NewController(GinkgoT())
+			gardenClient = mock.NewMockInterface(ctrl)
+			k8sGardenRuntimeClient = mockclient.NewMockClient(ctrl)
```

Comment:
The clientset (`kubernetes.Interface`) is just a container for all the different clients we use. If you just need it to pass down a runtime client in your test, you can use the fake clientset, which makes it easier to read/understand, e.g.:
https://github.com/gardener/gardener/blob/a39cd072a8c6eabab4c408a3e18c0b90b920e851/pkg/controllermanager/controller/controllerregistration/controllerregistration_control_test.go#L288-L289

Reply:
Yes was wondering if I should directly use the fake client, but copy pasted from the tests above :/, I'll see if I can improve the test a bit later today.

Reply:
It was just suggestion. But if you want to adapt, perfect :)